lib/pharos/phases/configure_cluster_info.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+require_relative 'base'
+
+module Pharos
+  module Phases
+    class ConfigureClusterInfo < Base
+      # @param master [Pharos::Configuration::Host]
+      def initialize(master)
+        @master = master
+      end
+
+      def client
+        @client ||= Pharos::Kube.client(@master.address)
+      end
+
+      def call
+        logger.info { "Updating public cluster-info ..." }
+        configmap = client.get_config_map('cluster-info', 'kube-public')
+        kubeconfig = YAML.safe_load(configmap.data['kubeconfig'])
```

Comment:
Using the `Pharos::YamlFile`:
```ruby
  kubeconfig = Pharos::YamlFile.new(
    StringIO.new(configmap.data['kubeconfig']),
    override_filename: "#{@master.address}:kubeconfig"
   ).load
```

(it should maybe have something like `Pharos::YamlFile.from_string(xyz, "#{master.address}:kubeconfig")` to avoid the StringIO)